google/resource_google_project.go

```diff
@@ -155,6 +155,13 @@ func resourceGoogleProjectRead(d *schema.ResourceData, meta interface{}) error {
 		return handleNotFoundError(err, d, fmt.Sprintf("Project %q", pid))
 	}
 
+	// If the project has been deleted from outside Terraform, remove it from state file.
+	if p.LifecycleState != "ACTIVE" {
+		log.Printf("[WARN] Removing project '%s' because the project is not active.", pid)
```

Comment:
Minor; printing the state as well might be useful for debugging.

Reply:
Good idea. Added.